### PR TITLE
Fixed the inconsistent behavior for accents in search

### PR DESCRIPTION
**Pull request by LingZhang22**
Currently, the characters with accents are not fully supported in the search functionality as described in [#6815](https://github.com/JabRef/jabref/issues/6815). This PR aims to solve it.

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/model/search/rules/SearchRule.java [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 
-    PdfSearchResults getFulltextResults(String query, BibEntry bibEntry);
+    public abstract boolean validateSearchStrings(String query);
 
-    boolean validateSearchStrings(String query);
+    // convert the given string to ASCII string
+    static String convertToASCII(String string) {
+        String normalizedString = Normalizer.normalize(string, Normalizer.Form.NFD);
+        return normalizedString.replaceAll("[^\\p{ASCII}]", "");
+    }
```
Comment (koppor) [inline src/main/java/org/jabref/model/search/rules/SearchRule.java:20]: Similar to @Siedlerchr's comment: Move such helper method to `StringUtils`.

There is IMHO need to convert `SearchRule` from an interface to a class.

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed an issue where opening the changelog from withing JabRef led to a 404 error [#8563](https://github.com/JabRef/jabref/issues/8563)
 - We fixed an issue where not all found unlinked local files were imported correctly due to some race condition. [#8444](https://github.com/JabRef/jabref/issues/8444)
 - We fixed an issue where Merge entries dialog exceeds screen boundaries.
+- We fixed an issue where accent search does not perform consistently. [#6815]([https://github.com/JabRef/jabref/issues/6815])
```
Comment (koppor) [inline CHANGELOG.md:45]: Please use correct markdown syntax

```suggestion
- We fixed an issue where accent search does not perform consistently. [#6815](https://github.com/JabRef/jabref/issues/6815)
```

**Review by koppor — CHANGES_REQUESTED** [review]
Please add test cases to https://github.com/JabRef/jabref/blob/main/src/test/java/org/jabref/model/strings/StringUtilTest.java to check whether [`stripAccents`](https://github.com/JabRef/jabref/blob/432d7430d57b59f2cc65c42e09f604e174b868a2/src/main/java/org/jabref/model/strings/StringUtil.java#L694) works.

Note that this method calls https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#stripAccents-java.lang.String-, which should be similar to the one of the JDK

**Comment by koppor** [comment]
Proposal for continuing:

Start from the `main` branch, add test cases. Then work on the new main; meanwhile we merged https://github.com/JabRef/jabref/pull/8636, which reorganizes to code a bit.

**Comment by LingZhang22** [comment]
> I just noticed that we have a method stripAccents in our StringUtil class org.jabref.model.strings; Does that work as well?

Thanks! I will use the stripAccents method.

File: src/main/java/org/jabref/model/search/rules/SearchRule.java (hunk 1 above)
Comment (LingZhang22) [reply src/main/java/org/jabref/model/search/rules/SearchRule.java:20]: Will do!

File: CHANGELOG.md (hunk 2 above)
Comment (LingZhang22) [reply CHANGELOG.md:45]: I deleted the "[]". Thanks!

**Comment by LingZhang22** [comment]
> Proposal for continuing:
> 
> Start from the `main` branch, add test cases. Then work on the new main; meanwhile we merged #8636, which reorganizes to code a bit.

Thanks for the info. I reset the branch and added the changes to the current main.

**Review by calixtus — APPROVED** [review]
Codeweise Looks good to me!

**Comment by Siedlerchr** [comment]
Thanks for the follow up! LGTM!

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**